clang/lib/Sema/SemaExprMember.cpp

```diff
@@ -434,8 +434,11 @@ CheckExtVectorComponent(Sema &S, QualType baseType, ExprValueKind &VK,
   if (!HalvingSwizzle && *compStr) {
     // We didn't get to the end of the string. This means the component names
     // didn't come from the same set *or* we encountered an illegal name.
-    S.Diag(OpLoc, diag::err_ext_vector_component_name_illegal)
-      << StringRef(compStr, 1) << SourceRange(CompLoc);
+    size_t Offset = compStr - CompName->getNameStart() + 1;
+    char Fmt[3] = {'\'', *compStr, '\''};
+    S.Diag(OpLoc.getLocWithOffset(Offset),
+           diag::err_ext_vector_component_name_illegal)
+        << StringRef(Fmt, 3) << SourceRange(CompLoc);
```

Comment:
Can someone explain how this works?  It looks like (both before and after) that we're treating the name length as being only 1 (and in this case, 3 after we're adding the single quote on both sides).  

BUT it seems in test 91 below that the 'wyx' works, but is 3 characters long.  Why can't we just print the entirety of the `IdentifierInfo` above?

Reply:
A quick debug shows the below test changes are a little confusing, as the 'bool' vectors dont' actually allow anything, so this just ends up diagnosing elsewhere.  So this change _IS_ just diagnosing a single letter.

For some reason that I don't quite understand, the access is allowed to be any combo of `x`,`y`, `z`, `w`, `r`, `g`,` b`, and `a`, with some attempt to be able to 'set' them.  So it seems we're trying to diagnose the first one that isn't one of those above letters, so the single letter thing seems fine, just confusing for obvious reasons :)

Reply:
This is the error path for when you try to mix different accessor types on openCL vector types, which allows you to access them [with this syntax](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#vector-components) but disallows mixing the different styles.

For example, trying to do something like `.rgbz` where `z` is the bad accessor--here we're mixing the `xyzw` access style with the `rgba` access style. This should result in an error, as evidenced by the [test(s) here](https://github.com/llvm/llvm-project/blob/d866005f6928a2a97e67866bedb26139d8cc27d9/clang/test/Sema/ext_vector_components.c#L46)